pkg/parcacol/ingest.go

```diff
@@ -106,21 +106,9 @@ func (ing Ingester) IngestProfile(ctx context.Context, ls labels.Labels, p *prof
 		return fmt.Errorf("failed to convert samples to buffer: %w", err)
 	}
 
-	buffer.Sort()
-
-	// This is necessary because sorting a buffer makes concurrent reading not
-	// safe as the internal pages are cyclically sorted at read time. Cloning
-	// executes the cyclic sort once and makes the resulting buffer safe for
-	// concurrent reading as it no longer has to perform the cyclic sorting at
-	// read time. This should probably be improved in the parquet library.
-	buffer, err = buffer.Clone()
```

Comment:
We no longer need this since we serialize the parquet buffer, so effectively it's already being cloned. Also, the sorting was moved to the buffer creation.